simplipy/api.py

```diff
@@ -306,6 +334,31 @@ def _wrap_request_method(
             )(self._async_api_request),
         )
 
+    def _wrap_media_method(
```

Comment:
It feels strange that we're effectively duplicating logic that we already have for `_wrap_request_method`.

Reply:
I can change ``_wrap_request_method()` to take a function argument; either _async_api_request or _async_media_request.  Would that be better?

Reply:
Made it reusable by passing in functions.